### ISSUE TITLE
Unable to pass multiple language codes to expand_address

### ISSUE THREAD

**Issue by awiede** [pnordahl/rust-postal#6]
When using the `expand_address` binding I am unable to pass more than one language code to the `ExpandAddressOptions`. I can pass a slice with only one language in it, but as soon as I pass a second value I get a segfault.

See example code [here](https://github.com/awiede/rustpostal-test/blob/master/src/main.rs), which yields the following results:

```
Testing:
Address: "Foo Str"
Language(s): ["en"]
Results: ["foo street", "foo strata unit"]
Testing:
Address: "Foo Str"
Language(s): ["de"]
Results: ["foo strasse"]
Testing:
Address: "Foo Str"
Language(s): ["de", "en"]
[1]    20143 segmentation fault  cargo run
```

Expected result (using PyPostal):

```python
In [1]: from postal.expand import expand_address

In [2]: expand_address("Foo Str", languages=["en"])
Out[2]: ['foo street', 'foo strata unit']

In [3]: expand_address("Foo Str", languages=["de"])
Out[3]: ['foo strasse']

In [4]: expand_address("Foo Str", languages=["de", "en"])
Out[4]: ['foo strasse', 'foo street', 'foo strata unit']
```

**Comment by pnordahl** [comment]
Thanks for reporting this, I'll take a look at it.

**Comment by pnordahl** [comment]
@awiede I won't be able to get a fix out until later as this proved difficult to get working with the existing slice API for `set_languages`, but here is a workaround (setting the languages directly on the underlying C struct, for your test code):

Replace `expand_options.set_languages(languages.as_slice());` with the following:

> let c_langs: Vec<CString> = languages
        .iter()
        .map(|l| CString::new(*l).unwrap())
        .collect();
    let mut ptrs: Vec<*const c_char> = c_langs.iter().map(|cs| cs.as_ptr()).collect();
    expand_options.opts.languages = ptrs.as_mut_ptr() as *mut *mut c_char;
    expand_options.opts.num_languages = (c_langs.len() as u64).try_into().unwrap();

**Comment by pnordahl** [comment]
@awiede fixed in 0.2.4 on crates.io.

Thanks again for the report!

**Comment by awiede** [comment]
🎉 Thanks for the quick turn around, @pnordahl!
